src/Hive.IO/Hive.IO/Building/Zone.cs

```diff
@@ -343,53 +376,152 @@ public Zone(rg.Brep zone_geometry, int index, double tolerance, string roomType,
         {
-            this.MonthlySolarGainsPerWindow = new double[Q_s_per_window.Length][];
+            this.SolarGainsPerWindowMonthly = new double[Q_s_per_window.Length][];
+            this.SolarGainsPerWindowHourly = new double[Q_s_per_window.Length][];
             for (int i = 0; i < Q_s_per_window.Length; i++)
             {
-                this.MonthlySolarGainsPerWindow[i] = new double[Q_s_per_window[i].Length];
-                for (int j = 0; j < Q_s_per_window[i].Length; j++)
-                    this.MonthlySolarGainsPerWindow[i][j] = Q_s_per_window[i][j];
+                this.SolarGainsPerWindowMonthly[i] = new double[Misc.MonthsPerYear];
+                if (Q_s_per_window[i].Length == Misc.HoursPerYear)
+                {
+                    this.SolarGainsPerWindowHourly[i] = new double[Misc.HoursPerYear];
+                    Q_s_per_window[i].CopyTo(this.SolarGainsPerWindowHourly[i],0);
+                    Misc.GetCumulativeMonthlyValue(Q_s_per_window[i]).CopyTo(this.SolarGainsPerWindowMonthly[i],0);
+                }else
+                {
+                    Q_s_per_window[i].CopyTo(this.SolarGainsPerWindowMonthly[i], 0);
+                }
             }
         }
 
+
         /// <summary>
-        /// Setting monthly energy demands of this zone. Loads have to be computed externally, e.g. with Hive.Core SIA380
+        /// Setting energy demands of this zone. Loads have to be computed externally, e.g. with Hive.Core SIA380
         /// </summary>
         /// <param name="heatingLoads"></param>
         /// <param name="dhwLoads"></param>
         /// <param name="coolingLoads"></param>
         /// <param name="electricityLoads"></param>
-        public void SetEnergyDemandsMonthly(double[] heatingLoads, double[] dhwLoads, double[] coolingLoads, double[] electricityLoads)
+        public void SetEnergyDemands(double[] heatingLoads, double[] dhwLoads, double[] coolingLoads, double[] electricityLoads)
         {
             this.HeatingLoadsMonthly = new double[Misc.MonthsPerYear];
             this.DHWLoadsMonthly = new double[Misc.MonthsPerYear];
             this.CoolingLoadsMonthly = new double[Misc.MonthsPerYear];
             this.ElectricityLoadsMonthly = new double[Misc.MonthsPerYear];
 
-            heatingLoads.CopyTo(this.HeatingLoadsMonthly, 0);
-            dhwLoads.CopyTo(this.DHWLoadsMonthly, 0);
-            coolingLoads.CopyTo(this.CoolingLoadsMonthly, 0);
-            electricityLoads.CopyTo(this.ElectricityLoadsMonthly, 0);
+            if (heatingLoads.Length == Misc.HoursPerYear)
+            {
+                this.HeatingLoadsHourly = new double[Misc.HoursPerYear];
+                heatingLoads.CopyTo(this.HeatingLoadsHourly,0);
+                Misc.GetCumulativeMonthlyValue(heatingLoads).CopyTo(this.HeatingLoadsMonthly, 0);
+            }
+            else
+            {
+                heatingLoads.CopyTo(this.HeatingLoadsMonthly, 0);
+            }
```

Comment:
DId the following not work, is it necessary to use CopyTo here?

```cs
            if (heatingLoads.Length == Misc.HoursPerYear)
            {
                this.HeatingLoadsHourly = heatingLoads;
                this.HeatingLoadsMonthly = Misc.GetCumulativeMonthlyValue(heatingLoads);
            }
            else
            {
                this.HeatingLoadsMonthly = heatingLoads;
            }

```

Reply:
I'd like to avoid referencing the arrays, because they come from a separate module (sia380.py). that's why I'm copying

Reply:
I think that C#, unlike python, passes by value to methods unless specifically determined with the `ref` keyword in the method arg.  So that's why I figured you could treat it as a value rather than referencing the original object.

Here's a [stackoverflow](https://stackoverflow.com/questions/8708632/passing-objects-by-reference-or-value-in-c-sharp) that seems to say its a bit more complicated than that so I dunno what the best approach is. But i feel in this case it will be passed by value since we are only assigning values to the properties in this class rather than altering the incoming arguments.

Reply:
I will approve changes anyways and you can decide which way you would rather do it, in any case its a refactoring rather than a change in functionality :)

Reply:
> I think that C#, unlike python, passes by value to methods unless specifically determined with the `ref` keyword in the method arg. So that's why I figured you could treat it as a value rather than referencing the original object.
> 
> Here's a [stackoverflow](https://stackoverflow.com/questions/8708632/passing-objects-by-reference-or-value-in-c-sharp) that seems to say its a bit more complicated than that so I dunno what the best approach is. But i feel in this case it will be passed by value since we are only assigning values to the properties in this class rather than altering the incoming arguments.

I'm confused. I've learned that objects, including arrays, are reference types in c#; and I've had some nasty bugs because of it: https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/arrays/passing-arrays-as-arguments

I'll check it out later and if it's rly by val i'll change it

Reply:
![image](https://user-images.githubusercontent.com/21927657/125837471-c348d4f6-afad-45e4-bc3e-33f149f35f24.png)

they are byref. or is my latin not good enough again?

I'll keep the `CopyTo` for now and lets keep this for refactoring

Reply:
Oh my bad I was the confused one.... I looked more into it. Yes you are right classes and other objects, including arrays, are reference types, so we should for sure keep copyto here.